processfamily/__init__.py

```diff
@@ -381,9 +366,155 @@ def _handle_response_line(self, line):
 CPU_AFFINITY_STRATEGY_CHILDREN_ONLY = 1
 CPU_AFFINITY_STRATEGY_PARENT_INCLUDED = 2
 
-CHILD_COMMS_STRATEGY_NONE = 0
-CHILD_COMMS_STRATEGY_PIPES_CLOSE = 1
-CHILD_COMMS_STRATEGY_PROCESSFAMILY_RPC_PROTOCOL = 2
+
+class NoCommsStrategy(ChildCommsStrategy):
+    MONITOR_STDOUT = False
+    CAN_WAIT_FOR_TERMINATE = False
+
+    @staticmethod
+    def get_popen_streams(echo_std_err):
+        """Returns kwargs for stdin, stdout and stderr to pass to subprocess.Popen"""
+        return {"stdin": None, "stdout": None, "stderr": None}
+
+class ClosePipesCommsStrategy(ChildCommsStrategy):
+    def stop_child(self, end_time):
+        p = self._process_instance
+        logger.info("Closing stdin for process %r", self)
+        try:
+            p.stdin.close()
+        except Exception as e:
+            logger.warning("Failed to close child process input stream with PID %s: %s\n%s", self.pid, e, _traceback_str())
+        yield
+        yield
```

Comment:
The backup plan is currently that the child process will be terminated when the parent exits.
We do have unit tests that check various scenarios, notably `test_parent_exit_child_locked_up` to make sure that this does happen.
Obviously a "no I really want you to quit" might be useful too, but then for what we're using this for I think we would really want a way to restart processes etc, which is really a new feature.
Our current model is to assume that if something goes really wrong it will be cleaned up on the next parent process restart